crates/bins/src/bin/datadog-static-analyzer-git-hook.rs

```diff
@@ -508,30 +538,69 @@ fn main() -> Result<()> {
             })
             .collect();
 
-        for secret_result in secrets_results {
-            let path = PathBuf::from(secret_result.filename);
-
-            for secret_match in secret_result.matches {
-                if let Some(lines) = modifications.get(&path) {
-                    if lines.contains(&secret_match.start.line) {
-                        println!(
-                            "{}",
-                            format_error(
-                                path.display().to_string().as_str(),
-                                secret_match.start.line,
-                                &secret_result.rule_name,
-                                IssueType::Secret,
-                            )
-                        );
-
-                        fail_for_secrets = true;
-                        continue;
-                    }
-                }
+        // Filter the results only based on the difff
+        secrets_for_diff = secrets_results
+            .iter()
+            .map(|rr| {
+                let mut new_result = rr.clone();
+                let filtered_violations = new_result
+                    .matches
+                    .into_iter()
+                    .filter(|v| {
+                        let path = PathBuf::from(&rr.filename);
+                        if let Some(lines) = modifications.get(&path) {
+                            lines.contains(&v.start.line)
+                        } else {
+                            false
+                        }
+                    })
+                    .collect();
+                new_result.matches = filtered_violations;
+                new_result
+            })
+            .collect();
```

Comment:
Same story as above, though in this case you'll need to make `secrets_results` `mut`

```rs
        secrets_results.iter_mut().for_each(|rr| {
            rr.matches.retain(|v| {
                let path = PathBuf::from(&rr.filename);
                if let Some(lines) = modifications.get(&path) {
                    lines.contains(&v.start.line)
                } else {
                    false
                }
            });
        });
```